### PR TITLE
ISSUE-25 - add support for html5 video poster images

### LINKED ISSUE #25: Add support for HTML5 video poster images
HTML5 <video> elements support a ```poster``` attribute to define a static image to display before the video loads. These images are often full HD frames and can be the largest images on the page.
We should add support to identify poster images and treat them as a normal image.

### PR DESCRIPTION
**Pull request by tabanliviu**
fixes #25

### CODE REVIEW

File: app/public.js [hunk 1]
```diff
@@ -309,8 +313,11 @@
     }
   }
 
+  // video has currentSrc which points to the video file so we need to ignore it
   function getUrl(element) {
-    if (element.currentSrc) {
+    if (element.poster) {
+      return element.poster;
+    } else if (element.tagName.toLowerCase() !== 'video' && element.currentSrc) {
```
Comment (MalcolmSmithUK) [inline app/public.js]: @tabanliviu KISS - This indicates that the previous condition isn't reliable enough, so we're pushing the decision elsewhere.  Perhaps it makes more sense to be explicit and inclusive? Are there any other situations where simply checking for presence of currentSrc may not be suitable?

```
if (element.tagName.toLowerCase() === 'video') {
  return !!element.poster;
}
else if (element.currentSrc) {
   ...
}
else {
  ...
}
```

File: test/public.spec.js [hunk 2]
```diff
@@ -91,7 +94,7 @@ describe('imageChecker', () => {
       window.NCC.imageChecker.showImagesInfo();
 
       let imageOverlays = document.querySelectorAll('.ncc-image-checker-overlay');
-      expect(imageOverlays.length).toEqual(5);
+      expect(imageOverlays.length).toEqual(6);
```
Comment (MalcolmSmithUK) [inline test/public.spec.js:90]: @tabanliviu INFO - So which condition added the extra overlay... `(empty)` or `{ poster: false }`?

**Review by MalcolmSmithUK — CHANGES_REQUESTED** [review]

**Comment by tabanliviu** [comment]
@MalcolmSmithUK updated with requested changes

File: test/public.spec.js (hunk 2 above)
Comment (tabanliviu) [reply test/public.spec.js:90]: @MalcolmSmithUK updated with requested changes

**Review by MalcolmSmithUK — APPROVED** [review]

**Pull request merged**